nextflow_schema.json

```diff
@@ -2,109 +2,130 @@
-                },
-                "blast_evalue": {
-                    "type": "number",
-                    "default": 0.1
+                "enable_filter": {
+                    "type": "boolean",
+                    "description": "If set to True the filter is used. Otherwise only assessing is performed."
                 },
-                "blast_identity": {
-                    "type": "number",
-                    "default": 90
+                "filter_trimmed": {
+                    "type": "boolean",
+                    "description": "If set to True the the preproccesed reads are used for filtering. Else the raw reads are used"
                 },
-                "blast_max_hsps": {
-                    "type": "integer",
-                    "default": 1
+                "skip_blastn": {
+                    "type": "boolean",
+                    "description": "Defines if blastn should be used or not. By default blastn is used in the workflow. If also the filter is enabled the reads are filtered using the output from blastn."
                 },
-                "blast_max_target_seqs": {
-                    "type": "integer",
-                    "default": 1
+                "filter_with_kraken2": {
+                    "type": "boolean",
+                    "description": "If this is set to True the kraken2 ouput is used for filtering."
                 }
             }
         },
-        "workflow_parameters": {
-            "title": "Workflow parameters",
+        "kraken2": {
+            "title": "kraken2",
             "type": "object",
-            "description": "",
+            "description": "Parameters used by kraken2 to classify all reads provided.",
             "default": "",
             "properties": {
-                "single_end": {
-                    "type": "boolean"
-                },
-                "long_read": {
-                    "type": "boolean"
-                },
-                "enable_filter": {
-                    "type": "boolean"
+                "kraken2db": {
+                    "type": "string",
+                    "default": "https://genome-idx.s3.amazonaws.com/kraken/k2_standard_08gb_20231009.tar.gz",
+                    "description": "The database which is used in the classification step. For input how to use this parameter to fine-tune the step refer to usage/Databases/kraken2"
                 },
-                "filter_trimmed": {
-                    "type": "boolean"
+                "save_output_fastqs": {
+                    "type": "boolean",
+                    "description": "If the unclassified and classified reads should be saved to different files. The default false should be used",
+                    "hidden": true
                 },
-                "skip_blastn": {
-                    "type": "boolean"
+                "save_reads_assignment": {
```

Comment:
A bit unconventional to have a default true flag, and may lead to confusion

Reply:
Needed by the nf-core kraken2 module. Any suggestions how to handle this in a better way?

Reply:
In https://github.com/jannikseidelQBiC/nf-core_detaxizer/blob/917db5448e10cb351bd43d4ac4236db29beef330/workflows/detaxizer.nf#L199-L204 use instead of
```
    KRAKEN2_KRAKEN2 (
        FASTP.out.reads,
        ch_kraken2_db,
        params.save_output_fastqs,
        params.save_reads_assignment
    )
```
something like
```
    KRAKEN2_KRAKEN2 (
        FASTP.out.reads,
        ch_kraken2_db,
        params.save_output_fastqs,
        true
    )
```